sos/report/plugins/authd.py

```diff
@@ -0,0 +1,53 @@
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.report.plugins import Plugin, UbuntuPlugin
+
+
+# See the docs: https://documentation.ubuntu.com/authd/
+class Authd(Plugin, UbuntuPlugin):
+
+    short_desc = 'Authd daemon & broker information'
+
+    plugin_name = 'authd'
+
+    packages = ('authd', 'authd-msentraid', 'authd-google')
+
+    def setup(self):
+        self.add_dir_listing([
+            "/etc/authd/brokers.d",
+        ])
+
+        self.add_copy_spec([
+            "/etc/authd/brokers.d/msentraid.conf",
+            "/etc/authd/brokers.d/google.conf",
+            "/var/snap/authd-google/current/broker.conf",
+            "/var/snap/authd-google/current/broker.conf.d/*",
+            "/var/snap/authd-msentraid/current/broker.conf",
+            "/var/snap/authd-msentraid/current/broker.conf.d/*",
+        ])
+
+    def postproc(self):
+        # Entra uses hex encoded IDs/secrets so just filter all hex data (with
+        # `-`) to be safe. These can be generated with uuidgen:
+        # $ uuidgen
+        # dd591ced-483e-4c47-beaf-ff46f68aab0a
+        self.do_path_regex_sub(
+            r".*",
+            r"[a-fA-F0-9-]{6,}",
```

Comment:
This will match also words like `de-caf` or `caffee` or `baffed` etc. - are you ok with obfuscating some unnecessary strings (even substrings inside a word)?

Reply:
Upped the quantifier to 18, that should reduce the likelihood of a false positive.